pkg/frr/frr.go

```diff
@@ -218,6 +322,12 @@ func handlevrf(objectData *eventbus.ObjectData) {
 			comp.CompStatus = common.ComponentStatusError
 		}
 		log.Printf("%+v\n", comp)
+
+		// Checking the timer to decide if we need to replay or not
+		if comp.Timer > replayThreshold {
```

Comment:
what exactly you mean by saying "only one check" ? Like creating a common function and calling it from both device types related functions ?

Reply:
yes

Reply:
Done

Reply:
Would it make sense to move it entirely into comp?
```go
func CheckReplayThreshold(replayThreshold time.Duration) {
	replay = (timer > replayThreshold)
}
```

Reply:
Sorry didn't get this one. Where you think we should move the function ? Can you please elaboate a bit ?

Reply:
Is it possible to move `CheckReplayThreshold` entirely into `comp` type? Then we don't need to pass the same `comp.Timer` all the time as well as assign the result to the same `comp.Replay`. That `comp` just have a method which handles all of that and work with its own members. Wondering if `replayThreshold` as well as this chunk of code
```go
               if status {
			comp.CompStatus = common.ComponentStatusSuccess
			comp.Timer = 0
		} else {
			if comp.Timer == 0 {
				comp.Timer = 2 * time.Second
			} else {
				comp.Timer *= 2
			}
			comp.CompStatus = common.ComponentStatusError
		}
```
can also go into `comp`?

Reply:
I think we can do many impovements here and we can do them in the experiment branch that I have created. The reason for this is:
1. This code snippet that you have posted above is common in all the modules and not just FRR. So changes in that code will touch all the components and exceeds the scope of this PR so let's do it better on the experiment branch to do it properly
2. The replayThreshold variable is decided by each module itself and that is why we cannot pass it to the comp structure that easily. If we want to pass it in that Comp structure then we need to find a clever way to do so. That means that it needs a bit more thinking and changes that will affect other part of the code that are outside of this scope. So let's do it to the experiment branch

The rest of your comments I have fixed

Reply:
> This code snippet that you have posted above is common in all the modules and not just FRR. So changes in that code will touch all the components and exceeds the scope of this PR so let's do it better on the experiment branch to do it properly

You just added it in this PR, why do you need to rework the rest?

> The replayThreshold variable is decided by each module itself and that is why we cannot pass it to the comp structure that easily.

It is the corresponding handle*() call which creates com, and can pass the threshold at construction?

I just do not want we continue adding debt. If it does not require to rework other components, so I'd prefer it is fixed in place
This change looks pretty straighforward

Reply:
> You just added it in this PR, why do you need to rework the rest?

I didnt added this code snippet to this PR that was allready there.

```if status {
			comp.CompStatus = common.ComponentStatusSuccess
			comp.Timer = 0
		} else {
			if comp.Timer == 0 {
				comp.Timer = 2 * time.Second
			} else {
				comp.Timer *= 2
			}
			comp.CompStatus = common.ComponentStatusError
		}


```

Reply:
Are you talking about this comp here ? https://github.com/opiproject/opi-evpn-bridge/blob/c9baa3ee74869d690c7c63751572252165071946/pkg/frr/frr.go#L163

This is not the same comp object which we use in order to call the CheckReplayThreshold function

The object that we use to call this function is created here: https://github.com/opiproject/opi-evpn-bridge/blob/c9baa3ee74869d690c7c63751572252165071946/pkg/frr/frr.go#L199 

an is a different object so even if we add the replay threshold to the object in the line 163 will not be there in the line 199 because we call a function with a different object

Reply:
That's why I am saying that we need to find a better smarter way to handle this in the refactoring. Also the modules in general have duplicate code that can be written better so let's handle this also in the refactoring. WDYT ?